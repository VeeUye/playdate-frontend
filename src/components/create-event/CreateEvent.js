import React, { useState, useEffect } from "react";
import { UserAuth } from "../../contexts/AuthContext";
import getUserFriends from "../../requests/users/getUserFriends";

import screenSize from "../../functions/screenSize";
import SmallTitle from "../atoms/small-title/SmallTitle";
import CreateEventForm from "./CreateEventForm";
import LoadSpinner from "../load-spinner/LoadSpinner";

import Image from "../../assets/images/float.svg";

import createEventStyles from "../create-event/create-event.module.css";
import titleStyles from "../atoms/small-title/small-title.module.css";
import "./create-event.module.css";

const CreateEvent = () => {
  const isSmall = screenSize();

  const { user, token } = UserAuth();

  const [friends, setFriends] = useState([]);

  useEffect(() => {
    getUserFriends(user.uid, token).then((result) => {
      if (result) {
        const friendsInvite = result.map((friend) => {
          return { value: friend.userId, label: friend.name };
        });
        setFriends(friendsInvite);
      }
    });
  }, [user]);

  return (
    <>
      {!user || !token || !friends ? (
        <LoadSpinner />
      ) : (
        <div
          className={
            isSmall
              ? createEventStyles.smallScreen
              : createEventStyles.bigScreen
          }
        >
          <div className="outer" data-testid="create-event-outer">
            <SmallTitle className={titleStyles.default} text="Create Event" />
            <CreateEventForm
              user={user}
              token={token}
              friends={friends}
              data-testid="create-event-form"
            />
            <img
              className={createEventStyles.img}
              src={Image}
              alt="girl floating"
            />
          </div>
          <img
            className={createEventStyles.img2}
            src={Image}
            alt="girl floating"
          />
        </div>
      )}
    </>
  );
};

export default CreateEvent;
