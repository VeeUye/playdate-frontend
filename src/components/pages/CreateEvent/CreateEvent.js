import React, { useState, useEffect } from "react";
import { UserAuth } from "../../../contexts/AuthContext";
import getUserFriends from "../../../requests/users/getUserFriends";

import screenSize from "../../../functions/screenSize";
import SmallTitle from "../../atoms/small-title/SmallTitle";
import CreateEventForm from "../../organisms/CreateEventForm";
import LoadSpinner from "../../load-spinner/LoadSpinner";

import Image from "../../../assets/images/float.svg";

import styles from "./create-event.module.css";
import titleStyles from "../../atoms/small-title/small-title.module.css";
import classNames from "classnames";

const CreateEvent = () => {
  const isSmall = screenSize();
  const { user, token } = UserAuth();

  const [friends, setFriends] = useState([{}]);

  useEffect(() => {
    getUserFriends(user.uid, token).then((result) => {
      if (result) {
        const currentFriends = result.map((friend) => {
          return { value: friend.userId, label: friend.name };
        });
        console.log(currentFriends);
        setFriends(currentFriends);
      }
    });
  }, [user]);

  return (
    <>
      {!user || !token || !friends ? (
        <LoadSpinner />
      ) : (
        <div
          className={classNames(
            { [styles.smallScreen]: isSmall },
            { [styles.bigScreen]: !isSmall }
          )}
          data-testid="create-event-outer"
        >
          <div>
            <SmallTitle
              className={classNames(titleStyles.default, styles.title)}
              text="Create Event"
            />

            <CreateEventForm
              user={user}
              token={token}
              friends={friends}
              data-testid="create-event-form"
              key="create-event-outer"
            />
          </div>
          <img className={styles.img} src={Image} alt="girl floating" />
          <img className={styles.img2} src={Image} alt="girl floating" />
        </div>
      )}
    </>
  );
};

export default CreateEvent;
