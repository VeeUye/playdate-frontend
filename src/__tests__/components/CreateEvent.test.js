import React from "react";
import { render, screen } from "@testing-library/react";
import CreateEvent from "../../components/create-event/CreateEvent";

describe("CreateEvent", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<CreateEvent />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct title", () => {
    render(<CreateEvent />);
    const title = screen.getByText(/Create Event/i);

    expect(title).toBeVisible();
  });
});

describe("Create Event Form", () => {
  xit("displays the Event Name field ", () => {
    render(<CreateEvent />);

    expect(screen.getByLabelText("Event Name")).toBeVisible();

    expect(screen.getByLabelText("Date")).toBeVisible();

    expect(screen.getByLabelText("Time")).toBeVisible();

    expect(screen.getByLabelText("Location")).toBeVisible();

    expect(screen.getByLabelText("Invite")).toBeVisible();
  });
});

describe("Create Event Button", () => {
  it("displays a sign in button", () => {
    render(<CreateEvent />);

    const signInButton = screen.getByRole("button", { name: /create event/i });

    expect(signInButton).toHaveTextContent(/create event/i);
  });
});
