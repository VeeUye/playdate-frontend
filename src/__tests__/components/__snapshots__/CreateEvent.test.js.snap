// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEvent renders correctly 1`] = `
<DocumentFragment>
  <div
    class="background"
  >
    <div
      class="default"
    >
      Create Event
    </div>
    <form>
      <div
        class="field1"
      >
        <div>
          <div
            class="field"
          >
            <label
              class="label"
              for="name"
            >
              Event Name
              <input
                class="input"
                id="name"
                name="name"
                type="text"
                value="test"
              />
            </label>
          </div>
          <div
            class="field"
          >
            <label
              class="label"
              for="date"
            >
              Date
              <input
                class="input"
                id="date"
                name="date"
                type="date"
                value=""
              />
            </label>
          </div>
          <div
            class="splitInput"
          >
            <div
              class="field"
            >
              <label
                class="label"
                for="startTime"
              >
                Start
                <input
                  class="input"
                  id="startTime"
                  name="startTime"
                  type="time"
                  value=""
                />
              </label>
            </div>
            <div
              class="field"
            >
              <label
                class="label"
                for="endTime"
              >
                End
                <input
                  class="input"
                  id="endTime"
                  name="endTime"
                  type="time"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            class="field"
          >
            <label
              class="label"
              for="location"
            >
              Location
              <input
                class="input"
                id="location"
                name="location"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="field"
          >
            <label
              class="label"
              for="invite"
            >
              Invite
              <input
                class="input"
                id="invite"
                name="invite"
                type="email"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
      <button
        class="createEvent"
      >
        Create Event
      </button>
    </form>
    <img
      alt="girl floating"
      class="img"
      src="float.svg"
    />
  </div>
</DocumentFragment>
`;
